### PR TITLE
vm-under-test/customize-vm: Refactor boot script

### PR DESCRIPTION
**Pull request by RamLavi**
This PR is changing the boot script so that it could work running on every boot.

### CODE REVIEW

**Comment by RamLavi** [comment]
Passed CI on CNV 4.16 cluster:
```
+ TEST_NAMESPACE=realtime-checkup-1
+ VM_UNDER_TEST_CONTAINER_DISK_IMAGE=quay.io/ramlavi/kubevirt-realtime-checkup-vm:latest
+ TEST_CHECKUP_IMAGE=quay.io/ramlavi/kubevirt-realtime-checkup:devel
+ make e2e-test
mkdir -p /home/ralavi/go/src/github.com/kiagnose/kubevirt-realtime-checkup/_go-cache
podman run --rm \
	-v /home/ralavi/go/src/github.com/kiagnose/kubevirt-realtime-checkup:/go/src/github.com/kiagnose/kubevirt-realtime-checkup:Z \
	-v /home/ralavi/go/src/github.com/kiagnose/kubevirt-realtime-checkup/_go-cache:/root/.cache/go-build:Z \
	-v /home/ralavi/.kube/sno03-cnvqe2-rdu2:/root/.kube:Z,ro \
	--workdir /go/src/github.com/kiagnose/kubevirt-realtime-checkup \
	-e KUBECONFIG=/root/.kube/kubeconfig \
	-e TEST_NAMESPACE=realtime-checkup-1 \
	-e TEST_CHECKUP_IMAGE=quay.io/ramlavi/kubevirt-realtime-checkup:devel \
	-e VM_UNDER_TEST_CONTAINER_DISK_IMAGE=quay.io/ramlavi/kubevirt-realtime-checkup-vm:latest \
	docker.io/library/golang:1.20.12 \
	go test -v ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h 
=== RUN   TestTests
Running Suite: Tests Suite - /go/src/github.com/kiagnose/kubevirt-realtime-checkup/tests
========================================================================================
Random Seed: 1711523182

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/go/src/github.com/kiagnose/kubevirt-realtime-checkup/tests/tests_suite_test.go:39
[BeforeSuite] PASSED [0.002 seconds]
------------------------------
Checkup execution should complete successfully
/go/src/github.com/kiagnose/kubevirt-realtime-checkup/tests/checkup_test.go:80
• [761.675 seconds]
------------------------------

Ran 1 of 1 Specs in 761.677 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestTests (761.68s)
PASS
ok  	github.com/kiagnose/kubevirt-realtime-checkup/tests	761.685s
```

logs:
```
2024/03/27 07:06:26 kubevirt-realtime-checkup starting...
2024/03/27 07:06:26 Using the following config:
2024/03/27 07:06:26 	"vmUnderTestTargetNodeName": ""
2024/03/27 07:06:26 	"vmUnderTestContainerDiskImage": "quay.io/ramlavi/kubevirt-realtime-checkup-vm:latest"
2024/03/27 07:06:26 	"oslatDuration": "10m0s"
2024/03/27 07:06:26 	"oslatLatencyThresholdMicroSeconds": "45µs"
2024/03/27 07:06:26 Creating ConfigMap "realtime-checkup-1/realtime-vm-config"...
W0327 07:06:26.964391       1 warnings.go:70] metadata.finalizers: "foregroundDeleteVirtualMachine": prefer a domain-qualified finalizer name to avoid accidental conflicts with other finalizer writers
2024/03/27 07:06:26 Waiting for VMI "realtime-checkup-1/realtime-vmi-under-test-dkp9t" to boot...
2024/03/27 07:08:26 VMI "realtime-checkup-1/realtime-vmi-under-test-dkp9t" had successfully booted
2024/03/27 07:08:26 Login to VMI under test...
2024/03/27 07:08:51 VMI under test guest kernel Args: cat /proc/cmdline
BOOT_IMAGE=(hd0,gpt2)/vmlinuz-4.18.0-348.rt7.130.el8.x86_64 root=UUID=a9332d7d-1762-41cd-a702-6b2cc556c248 ro console=tty0 rd_NO_PLYMOUTH crashkernel=auto resume=UUID=d4ac9572-e828-4c87-9b76-e59c9fa6e426 console=ttyS0,115200 skew_tick=1 tsc=reliable rcupdate.rcu_normal_after_boot=1 isolcpus=managed_irq,domain,2-3 intel_pstate=disable nosoftlockup nohz=on nohz_full=2-3 rcu_nocbs=2-3 irqaffinity=0,1
[root@realtime-vmi-under-test-dkp9t ~]# 
2024/03/27 07:08:51 Running Oslat test on VMI under test for 10m0s...
2024/03/27 07:18:53 Oslat test completed:
taskset -c 2-3 oslat --cpu-list 2-3 --rtprio 1 --duration 10m0s --workload memmove --workload-mem 4K 
oslat V 2.60
Total runtime: 		600 seconds
Thread priority: 	SCHED_FIFO:1
CPU list: 		2-3
CPU for main thread: 	0
Workload: 		memmove
Workload mem: 		4 (KiB)
Preheat cores: 		2

Pre-heat for 1 seconds...
Test starts...
Test completed.

        Core:	 2 3
Counter Freq:	 2096 2096 (MHz)
    001 (us):	 5093591146 5093590896
    002 (us):	 0 1
    003 (us):	 622 657
    004 (us):	 117 81
    005 (us):	 21 21
    006 (us):	 14 16
    007 (us):	 11 11
    008 (us):	 4 4
    009 (us):	 3 2
    010 (us):	 0 0
    011 (us):	 0 0
    012 (us):	 0 0
    013 (us):	 0 0
    014 (us):	 0 0
    015 (us):	 0 0
    016 (us):	 0 0
    017 (us):	 0 0
    018 (us):	 0 0
    019 (us):	 0 0
    020 (us):	 0 0
    021 (us):	 0 1
    022 (us):	 0 0
    023 (us):	 0 0
    024 (us):	 0 0
    025 (us):	 0 0
    026 (us):	 0 0
    027 (us):	 0 0
    028 (us):	 0 0
    029 (us):	 0 0
    030 (us):	 0 0
    031 (us):	 0 0
    032 (us):	 0 0 (including overflows)
     Minimum:	 0 0 (us)
     Average:	 1.000 1.000 (us)
     Maximum:	 8 20 (us)
     Max-Min:	 8 20 (us)
    Duration:	 599.734 599.734 (sec)

[root@realtime-vmi-under-test-dkp9t ~]# 
2024/03/27 07:18:53 Max Oslat Latency measured: 20µs
2024/03/27 07:18:53 Trying to delete VMI: "realtime-checkup-1/realtime-vmi-under-test-dkp9t"
2024/03/27 07:18:53 Deleting ConfigMap "realtime-checkup-1/realtime-vm-config"...
2024/03/27 07:18:53 Waiting for VMI "realtime-checkup-1/realtime-vmi-under-test-dkp9t" to be deleted...
2024/03/27 07:18:59 VMI "realtime-checkup-1/realtime-vmi-under-test-dkp9t" was deleted successfully

```

File: vms/vm-under-test/scripts/customize-vm [hunk 1]
```diff
@@ -90,21 +90,24 @@ cat <<'EOF' > "$checkup_boot_script_full_path"
 # Copyright 2024 Red Hat, Inc.
 #
 
-set -e
```
Comment (orelmisan) [inline vms/vm-under-test/scripts/customize-vm:93]: Commit message:
> vm-under-test/customize-vm: Add flag to run tuned-adm once

This line repeats itself twice.

File: vms/vm-under-test/scripts/customize-vm [hunk 2]
```diff
@@ -90,21 +90,24 @@ cat <<'EOF' > "$checkup_boot_script_full_path"
...
+set -x
```
Comment (orelmisan) [inline vms/vm-under-test/scripts/customize-vm:92]: Could you please state in the commit message what is the reason for this change?

File: vms/vm-under-test/scripts/customize-vm [hunk 3]
```diff
@@ -90,21 +90,24 @@ cat <<'EOF' > "$checkup_boot_script_full_path"
...
 
 checkup_boot_ready_marker_full_path=$1
+checkup_tuned_adm_set_marker_full_path="/var/realtime-checkup-tuned-adm-set-marker"
```
Comment (orelmisan) [inline vms/vm-under-test/scripts/customize-vm:94]: Is this path similar to the DPDK checkup?

File: vms/vm-under-test/scripts/customize-vm [hunk 4]
```diff
@@ -90,21 +90,24 @@ cat <<'EOF' > "$checkup_boot_script_full_path"
...
 
 if systemctl --type swap list-units | grep -q '.swap'; then
   systemctl mask "$(systemctl --type swap list-units | grep '.swap' | awk '{print $1}')"
 fi
 
-tuned_conf="/etc/tuned/realtime-virtual-guest-variables.conf"
-echo "isolated_cores=2-3" > "$tuned_conf"
-echo "isolate_managed_irq=Y" >> "$tuned_conf"
-tuned-adm profile realtime-virtual-guest
+if [ ! -f "$checkup_tuned_adm_set_marker_full_path" ]; then
```
Comment (orelmisan) [inline vms/vm-under-test/scripts/customize-vm:100]: Please consider reversing the condition so the script will `exist 0` in case the file exists.

By doing so, one indentation level will be eliminated.

File: vms/vm-under-test/scripts/customize-vm [hunk 5]
```diff
@@ -106,6 +106,7 @@ if [ ! -f "$checkup_tuned_adm_set_marker_full_path" ]; then
   tuned-adm profile realtime-virtual-guest
   touch $checkup_tuned_adm_set_marker_full_path
   reboot
+  exit 0
```
Comment (orelmisan) [inline vms/vm-under-test/scripts/customize-vm:107]: Please consider squashing it with the previous commit.

File: vms/vm-under-test/scripts/customize-vm [hunk 6]
```diff
@@ -109,6 +107,8 @@ if [ ! -f "$checkup_tuned_adm_set_marker_full_path" ]; then
   exit 0
 fi
 
+boot_script_readiness_marker_full_path="/tmp/realtime-checkup-ready-marker"
```
Comment (orelmisan) [inline vms/vm-under-test/scripts/customize-vm:110]: Since this variable is only used once, please consider in-lining it.

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @RamLavi 
Please see the inline comments.

File: vms/vm-under-test/scripts/customize-vm (hunk 4 above)
Comment (RamLavi) [reply vms/vm-under-test/scripts/customize-vm:100]: you still need to set the final flag (in the next commit) so you won't gain removing the indentation...
This way however keeps it consistent with the swap condition, so I think it's better..

File: vms/vm-under-test/scripts/customize-vm (hunk 6 above)
Comment (RamLavi) [reply vms/vm-under-test/scripts/customize-vm:110]: This variable is going to be moved to a const once we move it to cloud-init. Keeping it will make the "copy" easier to understand. But I won't insist.

File: vms/vm-under-test/scripts/customize-vm (hunk 3 above)
Comment (RamLavi) [reply vms/vm-under-test/scripts/customize-vm:94]: It is ([here](https://github.com/kiagnose/kubevirt-dpdk-checkup/blob/9ecc55e13b1aa5167ab25d5407650672739e2dc6/pkg/internal/config/config.go#L67))

File: vms/vm-under-test/scripts/customize-vm (hunk 5 above)
Comment (RamLavi) [reply vms/vm-under-test/scripts/customize-vm:107]: why? it's a small change but I think it's better understood and explained when it is in a separate commit.

File: vms/vm-under-test/scripts/customize-vm (hunk 1 above)
Comment (RamLavi) [reply vms/vm-under-test/scripts/customize-vm:93]: DONE

File: vms/vm-under-test/scripts/customize-vm (hunk 2 above)
Comment (RamLavi) [reply vms/vm-under-test/scripts/customize-vm:92]: DONE

Comment (orelmisan) [reply vms/vm-under-test/scripts/customize-vm:92]: The reason why `set -e` was removed is explained, but the reason why `set -x` is required is missing.

File: vms/vm-under-test/scripts/customize-vm [hunk 7]
```diff
@@ -90,21 +90,24 @@ cat <<'EOF' > "$checkup_boot_script_full_path"
 # Copyright 2024 Red Hat, Inc.
 #
 
-set -e
+set -x
 
 checkup_boot_ready_marker_full_path=$1
```
Comment (orelmisan) [inline vms/vm-under-test/scripts/customize-vm]: This variable is set but not used.

File: vms/vm-under-test/scripts/customize-vm (hunk 2 above)
Comment (RamLavi) [reply vms/vm-under-test/scripts/customize-vm:92]: Right, DONE

File: vms/vm-under-test/scripts/customize-vm (hunk 7 above)
Comment (RamLavi) [reply vms/vm-under-test/scripts/customize-vm]: Fixed in the first commit, later it is removed.

**Review by orelmisan — APPROVED** [review]
Thank you
